### PR TITLE
Feature/react battle list

### PR DESCRIPTION
**Pull request by DeborahAlmeida**
## Objective and Motivation
- Create page to render battle list
- Use redux to get battle's data

### CODE REVIEW

File: backend/templates/spa/spa_template.html [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
   {% if user.is_authenticated %}
     <div class="user">
-        <a class ="anchor_user" href="{% url 'battle_list' %}">
+        <a class ="anchor_user" href="{% url 'spa_template_list' %}">
```
Comment (rsarai) [inline backend/templates/spa/spa_template.html]: this namespace can be improved :eyes:

File: frontend/js/App.js [hunk 2]
```diff
@@ -11,6 +12,7 @@ const App = () => (
     <Router>
       <Switch>
         <Route component={BattleDetail} path={Urls.spa_template(':id')} />
+        <Route component={BattleList} path={Urls.spa_template_list()} />
```
Comment (rsarai) [inline frontend/js/App.js]: same here

File: frontend/js/actions/setBattle.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { getTeamData } from 'utils/api';
```
Comment (rsarai) [inline frontend/js/actions/setBattle.js]: let's define a pattern here, to use `set` or `get`

File: frontend/js/actions/setBattle.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import { BATTLE_DETAIL } from '../constants';
+
+function fetchBattle(battle) {
```
Comment (rsarai) [inline frontend/js/actions/setBattle.js]: this is the battle or the battle id? If it's the battle id let's make it clear

File: frontend/js/actions/getBattleList.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { getBattles } from 'utils/api';
```
Comment (rsarai) [inline frontend/js/actions/getBattleList.js:1]: the name of this action is different out of the pattern here, let's define a pattern or set or get

File: frontend/js/actions/getBattleList.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+import { BATTLE_LIST } from '../constants';
+
+function fetchBattleList() {
```
Comment (rsarai) [inline frontend/js/actions/getBattleList.js]: The same goes for the function names on all actions

File: frontend/js/pages/BattleList.js [hunk 7]
```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+
+import { fetchBattleList } from '../actions/getBattleList';
+import { setCurrentUser } from '../actions/setUser';
+import Urls from '../utils/urls';
+
+function BattleList(props) {
+  useEffect(() => {
+    props.setCurrentUser();
```
Comment (rsarai) [inline frontend/js/pages/BattleList.js]: What happens if the user already is saved on the store?

File: frontend/js/reducers/index.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { combineReducers } from 'redux';
+
+import { battleListReducer } from './battleListReducer';
+import { battleReducer } from './battleReducer';
+import { userReducer } from './userReducer';
+
+export const Reducers = combineReducers({
+  battleState: battleReducer,
+  userState: userReducer,
+  battleListState: battleListReducer,
```
Comment (rsarai) [inline frontend/js/reducers/index.js]: please, rename this to `battles`

File: frontend/js/reducers/battleReducer.js [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { BATTLE_DETAIL } from '../constants';
```
Comment (rsarai) [inline frontend/js/reducers/battleReducer.js]: I already reviewed this and some other parts of this PR, did you have any branch issues?

File: frontend/js/reducers/battleListReducer.js [hunk 10]
```diff
@@ -0,0 +1,10 @@
+import { BATTLE_LIST } from '../constants';
+
+export const battleListReducer = (state = {}, action) => {
```
Comment (rsarai) [inline frontend/js/reducers/battleListReducer.js]: A note, you can have a single reducer for battle that deals with multiple actions. Please move this to the `battleReducer.js`

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/templates/spa/spa_template.html (hunk 1 above)
Comment (DeborahAlmeida) [reply backend/templates/spa/spa_template.html]: thanks!

File: frontend/js/actions/setBattle.js (hunk 3 above)
Comment (DeborahAlmeida) [reply frontend/js/actions/setBattle.js]: great!

File: frontend/js/actions/setBattle.js (hunk 4 above)
Comment (DeborahAlmeida) [reply frontend/js/actions/setBattle.js]: I will change this to battleId, thanks!

File: frontend/js/actions/getBattleList.js (hunk 5 above)
Comment (DeborahAlmeida) [reply frontend/js/actions/getBattleList.js:1]: okay!

File: frontend/js/actions/getBattleList.js (hunk 6 above)
Comment (DeborahAlmeida) [reply frontend/js/actions/getBattleList.js]: Alright :dancers:

File: frontend/js/pages/BattleList.js (hunk 7 above)
Comment (DeborahAlmeida) [reply frontend/js/pages/BattleList.js]: I will create a conditional to just make API request if the user is still null

File: frontend/js/reducers/index.js (hunk 8 above)
Comment (DeborahAlmeida) [reply frontend/js/reducers/index.js]: great!

File: frontend/js/reducers/battleReducer.js (hunk 9 above)
Comment (DeborahAlmeida) [reply frontend/js/reducers/battleReducer.js]: I didn't, I guess

File: frontend/js/reducers/battleListReducer.js (hunk 10 above)
Comment (DeborahAlmeida) [reply frontend/js/reducers/battleListReducer.js]: great! Thanks :)

File: frontend/js/pages/BattleList.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+
+import { getBattleList } from '../actions/getBattleList';
+import { getCurrentUser } from '../actions/getUser';
+import Urls from '../utils/urls';
+
+function BattleList(props) {
+  const { user } = props;
```
Comment (rsarai) [inline frontend/js/pages/BattleList.js:9]: You can deconstruct this is a single line:
```javascript
const { user, battles } = props;
```

**Review by rsarai — APPROVED** [review]
requested one minor thing, but looks good overall

**Comment by rsarai** [comment]
There are some conflicts here, please fix them before merging the PR

**Pull request merged**